### ISSUE TITLE
Preferred way to update ghost values

### ISSUE THREAD

**Issue by masterleinad** [dealii/dealii#2857]
In the documentation for [Parallel computing with multiple processors using distributed memory](http://www.dealii.org/developer/doxygen/deal.II/group__distributed.html) we say 
> [...]
>Both the PETScWrappers::MPI::Vector and TrilinosWrappers::MPI::Vector class support specifying >this information (see step-40 and step-32, respectively) through the >PETScWrappers::MPI::Vector::update_ghost_values() function or, in the case of Trilinos, >construction of a vector with an the locally relevant degrees of freedom index set.
> [..]
Interestingly, `PETScWrappers::MPI::Vector` is the only `Vector` class that does not implement `update_ghost_values`, but in the documentation of the other `update_ghost_values` functions this is always referenced. So far we updated ghost values using `operator=` instead and I would have just created a PR saying in the documentation that this should be used and remove `update_ghost_values`.

However, `LinearAlgebra::distributed::Vector< Number >::update_ghost_values` now implements this class with something useful. I guess the idea would then be to have `update_ghost_values` (doing nothing) in the other `Vector` classes for conformity with `LinearAlgebra::distributed::Vector< Number >::update_ghost_values`. 

In the documentation we would still refer to `operator=` as this is the only way that works with all the parallel `Vector` classes. Any different opinions?

**Comment by kronbichler** [comment]
In #2535 we discussed the general layout of the deal.II distributed vector which also includes the `update_ghost_values`. @Rombur and I came to the conclusion that we plan to remove that function (at that point, also the documentation should be updated) and rather go through some lightweight conversion between the distributed vector and a `ReadWriteVector` without copying the data but doing the operation that corresponds to `update_ghost_values`. One thing why I haven't done this yet is because I need some time to think about the design of a `BlockVector` with in-place conversions. The problematic issue is that our main use case of block vectors is for linear algebra operations such as block preconditioners. However, at the point of vector modifcation such as assembly we'll have the `ReadWriteVector` and I don't really see a point in having a block vector of those. So the ReadWriteVector would need to know about the block vector internally, which complicates the design and takes some time to write...

**Comment by kronbichler** [comment]
That said, this should definitely addressed properly for the next release because we need to avoid to change interfaces too often.

**Comment by bangerth** [comment]
Should we at least update the documentation? It seems to me that the preferred solution is indeed to use `operator=`, so we could just replace the comment pointed out by saying so.

**Comment by masterleinad** [comment]
In my opinion this is at least temporarily a good idea.

**Comment by kronbichler** [comment]
I don't think we plan to do anything on this right now, adjusting target milestone.
